Neos.ContentRepository.Core/Classes/Feature/RebaseableCommand.php

```diff
@@ -2,34 +2,61 @@
 
 declare(strict_types=1);
 
-namespace Neos\ContentRepository\Core\Feature\Common;
+namespace Neos\ContentRepository\Core\Feature;
 
-/*
- * This file is part of the Neos.ContentRepository package.
- *
- * (c) Contributors of the Neos Project - www.neos.io
- *
- * This package is Open Source Software. For the full copyright and license
- * information, please view the LICENSE file which was distributed with this
- * source code.
- */
-
-use Neos\ContentRepository\Core\CommandHandler\CommandInterface;
 use Neos\ContentRepository\Core\EventStore\DecoratedEvent;
 use Neos\ContentRepository\Core\EventStore\Events;
+use Neos\ContentRepository\Core\EventStore\InitiatingEventMetadata;
+use Neos\ContentRepository\Core\Feature\Common\PublishableToWorkspaceInterface;
+use Neos\ContentRepository\Core\Feature\Common\RebasableToOtherWorkspaceInterface;
 use Neos\EventStore\Model\Event\EventId;
 use Neos\EventStore\Model\Event\EventMetadata;
+use Neos\EventStore\Model\Event\SequenceNumber;
 
 /**
- * Stores the command in the event's metadata for events on a content stream. This is an important prerequisite
- * for the rebase functionality-
- *
  * @internal
  */
-final class NodeAggregateEventPublisher
+final readonly class RebaseableCommand
```

Comment:
utility to pass the "extracted" commands from the event stream around and also enrich them...

... because the use of `NodeAggregateEventPublisher` is incorrect now as we need to enrich also dimension changed events ...

Also by extraction and enrichment into one place we solve this todo :)

> TODO: Add this logic to the NodeAggregateCommandHandler;
> so that we can be sure these can be parsed again.


see https://github.com/neos/neos-development-collection/pull/5301/commits/56fd230780dd0c23c3308ab1f8a0b250c1aa29f0 and https://github.com/neos/neos-development-collection/pull/5301/commits/f7c39ec64e0392b51b89321369dc204e71f6a80e

Reply:
Very nice